test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -2068,6 +2119,17 @@ shrinkAction'WithVars _ctx _st a = case a of
       | tableVars' <- QC.liftShrink (const []) tableVars
       ]
 
+    SupplyUnionCredits tableVar (R.UnionCredits x) -> [
+        Some $ SupplyUnionCredits tableVar (R.UnionCredits x')
+      | x' <- QC.shrink x
+      ]
+
+    SupplyPortionOfDebt tableVar (Portion x) -> [
+        Some $ SupplyPortionOfDebt tableVar (Portion x')
+      | x' <- QC.shrink x
+      , x' > 0
+      ]
```

Comment:
Interesting, this shrinks towards larger portions. E.g. supplying 1/3 of the debt shrinks to 1/2, which is counter-intuitive, but kind of makes sense. The other direction is probably trickier to get right.

We could also try shrinking to some absolute `SupplyUnionCredits`. The number of credits is kind of arbitrary, but it can't result in a loop and might be worth a try.

Reply:
I did both! But I'm preferring shrinking to `SupplyUnionCredits` to shrinking towards larger portions. The former is more likely to actually provide us with a minimal counterexample, I thinks. Larger portions implies more work being done